docs/advanced/signal_generators.md

```diff
@@ -0,0 +1,638 @@
+
+`generate_burst()`  writes a trigger to the signal generator, initiating the generation of a burst of waveforms.
+
+`get_waveform_constraints()` will return a series of ranges that a waveform's parameters must
+be within to be generated. These constraints can be used before generating a function to
+make sure that the parameters you will be supplying
+are not outside the bounds. The method only requires the desired waveform function (except on \[AWGs\](default: AWG)) to be provided.
+However, different aspects may need to be provided to get a more accurate representation of what can actually be generated.
+If no other inputs are provided, the smallest range of the attribute is chosen.
+An \[AWG's\](default: AWG) signal path affects the range of the offset and amplitude.
+Higher frequencies on \[AFGs\](default: AFG) will lower the upper bound of the amplitude,
+alongside the which impedance is set.
+
+`set_waveform_properties()` is functionally identical to `generate_function()`, but does not turn the
+source channel off or on, nor will it stop or start an <AWG:>.
+
+## Signal Generators
+
+An overview of the different signal generators which are available in `tm_devices`.
+This includes:
+
+- Features unique to each which should be kept in mind.
+- Constraints for each waveform parameter.
+
+---
+
+### TekScope Internal Arbitrary Function Generators
+
+```{autoclasstree} tm_devices.drivers.pi.scopes.tekscope
+---
+full:
+namespace: tm_devices.drivers.pi.scopes.tekscope
+align: center
+alt: Tekscope `IAFG` Class Diagram
+---
+```
+
+The TekScope series instruments are signal generators focused on waveform generation which operate on the Windows operating systems.
+They accept communication through <USB:> and <TCPIP:> interfaces.
+
+Requesting function generation on an <IAFG:> will initially turn it off. The frequency, offset,
+function, impedance, and amplitude are set in the stated order.
+If the function is a `SQUARE` wave, the duty cycle is used
+to set how long the pulses are. Symmetry decides what direction the `RAMP` function
+skews. After all parameters are set, the source channel is then turned back on.
+
+Setting up bursts of the <IAFG:> involves setting it to burst mode and loading in a specified number of bursts.
+
+IAFGs have access to the following functions, listed within
+[`SignalGeneratorFunctionsIAFG`][tm_devices.helpers.enums.SignalGeneratorFunctionsIAFG]:
+`SIN`, `SQUARE`, `RAMP`, `PULSE`, `PRNOISE`, `DC`, `SINC`, `GAUSSIAN`, `LORENTZ`, `ERISE`, `EDECAY`, `HAVERSINE`, `CARDIAC`, `ARBITRARY`
+
+```{note}
+[IAFGs](default: IAFG) are only accessible if the oscilloscope has the `AFG` license installed.
+```
+
+```{note}
+[IAFGs](default: IAFG) contain no waveform list, editable memory or user defined waveforms. This means arbitrary waveforms
+must be loaded from the hard drive.
+```
+
+```{note}
+Some functions, like `SINC`, `GAUSSIAN`, `LORENTZ`, `ERISE`, `EDECAY` and `HAVERSINE` already have an inbuilt offset.
+```
+
+```{note}
```

Comment:
All of these notes/hints/warnings/cautions are using Sphinx syntax. They need to be updated to use mkdocs syntax.